src/bp/log_bp.py

```diff
@@ -0,0 +1,115 @@
+import errno
+import logging
+import os
+from flask import Blueprint, request
+from .err_bp import handle_bad_input_501
+from .home_bp import set_msg, validate_json_keys
+from .sessions_bp import session_id_validation
+
+def log_level_validation(cur_log_level):
+    curr_log_lvls = ["CRITICAL", "ERROR", "WARNING", "INFO", "DEBUG", "NOTSET"]
+    return True if cur_log_level in curr_log_lvls else False
+
+
+# dynamic methods, each for the appropriate level:
+def print_CRITICAL(logger, logging_data):
+    for data in logging_data: logger.critical(data)
+
+def print_ERROR(logger, logging_data):
+    for data in logging_data: logger.error(data)
+
+def print_WARNING(logger, logging_data):
+    for data in logging_data: logger.warning(data)
+
+def print_INFO(logger, logging_data):
+    for data in logging_data: logger.info(data)
+
+def print_DEBUG(logger, logging_data):
+    for data in logging_data: logger.debug(data)
+
+def print_NOTSET(logger, logging_data):
+    for data in logging_data: logger.notset(data)
+
+
+# logging the data to a .log file:
+def log_data(logger, session_id, json_data):
```

Comment:
Add a unit test (+error oriented) for this function. Heavily used

Reply:
There is a test for this - I'll add an error oriented (missing)

Reply:
Let's see the tests today